core/src/avm2.rs

```diff
@@ -78,6 +78,9 @@ pub struct Avm2<'gc> {
     /// System classes.
     system_classes: Option<SystemClasses<'gc>>,
 
+    #[collect(require_static)]
+    native_table: FnvHashMap<u32, NativeMethodImpl>,
```

Comment:
What if it was a
```cpp
const NATIVE_TABLE: &[Option<NativeMethodImpl>] = [...generated...];
```
?
With the Option being for the "holes" caused by non-native methods. I'm assuming a huge chunk of methods will be native, so this shouldn't waste a lot of space, right?

Reply:
Alternatively, if we wanna stay with a hashmap, it can be made const with:
https://crates.io/crates/phf

Reply:
I think adding an extra dependency for this is overkill. I'll convert it to an array.